tests/foreman/api/test_syncplan.py

```diff
@@ -1107,15 +1107,13 @@ def test_positive_synchronize_custom_product_weekly_recurrence(self):
             self.validate_task_status(repo.id, max_tries=2)
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'], after_sync=False)
-        # Wait the rest of expected time
+        # Update with the current UTC time plus a delay
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
+        sync_plan.update(['sync_date'])
+        # Wait some of the delay time
         self.logger.info('Waiting {0} seconds to check product {1}'
                          ' was synced'.format(delay, product.name))
         sleep(delay * 3/4)
```

Comment:
I would drop that sleep altogether. The same goes for `sleep(delay/4)` at line 1105 (I can't comment on it here in Github...).

Let `validate_task_status` do the job for us. It will wait for up to 10 minutes.

Reply:
OK. I liked the logger messages. But that's just me :)